python/pyspark/ml/feature.py

```diff
@@ -2082,10 +2144,19 @@ def _create_model(self, java_model):
         """
         Private method to convert the java_model to a Python model.
         """
-        return Bucketizer(splits=list(java_model.getSplits()),
-                          inputCol=self.getInputCol(),
-                          outputCol=self.getOutputCol(),
-                          handleInvalid=self.getHandleInvalid())
+        if (self.isSet(self.inputCol)):
+            return Bucketizer(splits=list(java_model.getSplits()),
+                              inputCol=self.getInputCol(),
+                              outputCol=self.getOutputCol(),
+                              handleInvalid=self.getHandleInvalid())
+        else:
+            splitsArrayList = []
+            for x in list(java_model.getSplitsArray()):
+                splitsArrayList.append(list(x))
```

Comment:
what about `splitsArrayList = [list(x) for x in list(java_model.getSplitsArray())]`?